### DISCUSSION TITLE
[2023-07-24] GUI Weekly Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#7379] [Weekly Demo Day]
### Finished and merged the new sorting of components.

On this occasion, I also updated entry labels.

https://github.com/enso-org/enso/assets/3919101/4aead1e2-de01-4b6e-aa12-403af0b3c677

### Components group may be defined by tag.

You may add `GROUP` tag to the entry's docstring, and if it's followed by an existing group name, the entry is put to that group. 

For example, here is how you can add an entry to "Web" group:
![image](https://github.com/enso-org/enso/assets/3919101/24a7e8f4-b7a8-4e65-a72f-d4c4ed9b088e)
If the group is defined in another project, you need to specify its name:
![image](https://github.com/enso-org/enso/assets/3919101/9d937751-0285-497a-846f-9c91f43c3c91)

### New Component Browser panels.

Panels were moved around and icons are added to match the design. Now, you may also hide the documentation panel.

https://github.com/enso-org/enso/assets/6566674/409bebb5-572a-4760-852d-f666124689a2

Also, the documentation panel was updated, and breadcrumbs were put in the proper place.

https://github.com/enso-org/enso/assets/6566674/bc53d94e-c958-4bda-920d-70a1a8cfcb4e

### Component Browser Breadcrumbs.

After finishing and fixing PR with new look of CB breadcrumbs (presented last time) Michael was working on the integration with the documentation panel - in the new design the breadcrumbs should navigate documentation view only. Still in progress.

**Comment by wdanilo** [comment]
I was working mainly on cloud designs. In particular, the permissions sharing panel, the context menu, and improvements to Graph Editor (the improvements are not finished yet, but are almost there). I was also working on hiring and planning with Sylwia and other people. Here are a few screenshots from the cloud env:

<img width="1441" alt="image" src="https://github.com/enso-org/enso/assets/1623053/72d663f5-e08b-462a-8651-301c460c12d5">

<img width="1441" alt="image" src="https://github.com/enso-org/enso/assets/1623053/4fa2116f-02b4-4461-86bc-c905f8546e08">

<img width="1441" alt="image" src="https://github.com/enso-org/enso/assets/1623053/73dbfbbf-d8d6-4369-8158-1a660c9cd2e4">

<img width="1441" alt="image" src="https://github.com/enso-org/enso/assets/1623053/560dcaf0-d0a2-4105-ad6b-0874896772eb">

**Comment by kazcw** [comment]
- Updated CB Focus logic for new CB style, and implemented a set of additional focus behaviors. Merging this morning after small changes.
- Integrated new fuzzy matching with refactored CB. Opening PR after meeting.

**Comment by mwu-tow** [comment]
Last week updates:
* Notification API (#7202) — reviewed and tentatively ready for merge.
* Quite much CI stuff (uploading test reports, incremental builds, Graal-bump-related issues, #7375).

Further plans:
* Integrating Notification API and using it in place of our statusbar (#7373).
* Placeholders for binary operators (#7107).

**Comment by JaroslavTulach** [comment]
Engine work:
- #7324
- new benchmarks as part of #7348
- some speedup by #7369
- otherwise non-performance work

**Comment by Frizi** [comment]
Finished new node design. PR got reviewed, but still needs QA. https://github.com/enso-org/enso/pull/7311
- Added displaying argument names in prefix chains
- Finished dropdown widget hover behavior, also works properly in list editors now.
- Fixed font rendering to avoid occasional thin line artifacts at the sprite borders.
- Implemented proper port hover behavior for access chains. It is no longer possible to drop an edge on a dot or create invalid expression in part of the chain.

https://github.com/enso-org/enso/assets/919491/aa687e53-a2fa-4e95-a15f-132c05e6337a
